bigquery/table.go

```diff
@@ -232,12 +232,19 @@ const (
 
 	// HourPartitioningType uses an hour-based interval for time partitioning.
 	HourPartitioningType TimePartitioningType = "HOUR"
+
+	// MonthPartitioningType uses a month-based interval for time partitioning.
+	MonthPartitioningType TimePartitioningType = "MONTH"
+
+	// YearPartitioningType uses a year-based interval for time partitioning.
+	YearPartitioningType TimePartitioningType = "YEAR"
 )
 
 // TimePartitioning describes the time-based date partitioning on a table.
 // For more information see: https://cloud.google.com/bigquery/docs/creating-partitioned-tables.
 type TimePartitioning struct {
-	// Defines the partition interval type.  Supported values are "DAY" or "HOUR".
+	// Defines the partition interval type.  Supported values are "HOUR", "DAY", "MONTH", and "YEAR".
+	// When the interval type is not specified, default behavior is DAY.
```

Comment:
Should this be tested? Or is this existing behavior.

Reply:
undocumented behavior.  internally is b/155198865